production-planning/src/test/scala/dev.atedeg.mdm.productionplanning/Tests.scala

```diff
@@ -0,0 +1,98 @@
+import dev.atedeg.mdm.products.{ CheeseType, Product }
+import dev.atedeg.mdm.utils.*
+import dev.atedeg.mdm.utils.given
+import dev.atedeg.mdm.utils.monads.*
+
+trait Mocks {
+
+  val cheeseTypeRipeningDays: CheeseTypeRipeningDays = Map(
+    CheeseType.Squacquerone -> RipeningDays(4),
+    CheeseType.Ricotta -> RipeningDays(0),
+    CheeseType.Caciotta -> RipeningDays(8),
+    CheeseType.Casatella -> RipeningDays(4),
+    CheeseType.Stracchino -> RipeningDays(5),
+  )
+
+  val prodToProd1: ProductToProduce = ProductToProduce(Product.Caciotta(500), Quantity(5))
+  val prodToProd2: ProductToProduce = ProductToProduce(Product.Casatella(300), Quantity(10))
+  val prodToProd3: ProductToProduce = ProductToProduce(Product.Squacquerone(250), Quantity(10))
+
+  val orderedProd1: OrderedProduct = OrderedProduct(Product.Caciotta(500), Quantity(5))
+  val orderedProd2: OrderedProduct = OrderedProduct(Product.Casatella(300), Quantity(10))
+  val orderedProd3: OrderedProduct = OrderedProduct(Product.Squacquerone(250), Quantity(10))
+}
+
+@SuppressWarnings(Array("org.wartremover.warts.Any"))
+class Tests extends AnyFeatureSpec with GivenWhenThen with Matchers with Mocks {
+
+  Feature("Create the production plan for the day") {
+    Scenario("Raffaella wants to create the production plan") {
+      Given("a list of orders with deadline in 10 days")
+      val requiredBy = java.time.LocalDate.now.plusDays(10)
+      val orderedProducts = NonEmptyList.of(orderedProd1, orderedProd2, orderedProd3)
+      val orders = List(
+        Order(OrderID(UUID.randomUUID()), requiredBy, orderedProducts),
+        Order(OrderID(UUID.randomUUID()), requiredBy, orderedProducts),
+        Order(OrderID(UUID.randomUUID()), requiredBy, orderedProducts),
+      )
+      And("the production plan of the previous year for the same day")
+      val productsToProduce = NonEmptyList.of(prodToProd1, prodToProd2, prodToProd3)
+      val previousProductionPlan = ProductionPlan(productsToProduce)
+      And("an empty stock")
+      val stock: Stock = _ => StockedQuantity(0)
+      When("creating the production plan")
+      val productionPlanCreation: SafeActionTwoEvents[ProductionPlanReady, OrderDelayed, ProductionPlan] =
+        createProductionPlan(stock, cheeseTypeRipeningDays)(previousProductionPlan, orders)
+      val (events1, events2, productionPlan) = productionPlanCreation.execute
+      Then(
+        "the result is the mocked result of a magic AI products to produce estimator: " +
+          "NonEmptyList.of(ProductToProduce(Product.Caciotta(500), Quantity(5)))",
+      )
+      val todaysProductionPlan: ProductionPlan = ProductionPlan(
+        NonEmptyList.of(
+          ProductToProduce(Product.Caciotta(500), Quantity(5)),
+        ),
+      )
+      productionPlan shouldBe todaysProductionPlan
+      events1 should not be empty
+      events1.map(_.productionPlan) should contain(productionPlan)
+      events2 shouldBe empty
```

Comment:
We must not test that the estimate is an exact value, since the estimator is an heuristic/AI/mock it will change over time and not be always exactly what we would expect: this test would break in mysterious ways and does not add any guarantee. 
We just have to test that the behaviour is the expected one, that is:
- it emits the events for the delayed orders when some orders have to be delayed
- it emits a production plan ready event
- the return value is the same production plan emitted in the event

In no way should we inspect the actual estimated production plan value